sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/CertificateUtil.java

```diff
@@ -54,24 +56,31 @@ public static PrivateKey privateKeyFromPem(byte[] pem) {
     }
 
     /**
-     * Extracts the X509Certificate certificate from a PEM certificate.
+     * Extracts the X509Certificate certificate/certificate-chain from a PEM certificate.
      * @param pem the contents of a PEM certificate.
-     * @return the X509Certificate certificate
+     * @return the {@link List} of X509Certificate certificate
      */
-    public static X509Certificate publicKeyFromPem(byte[] pem) {
-        Pattern pattern = Pattern.compile("(?s)-----BEGIN CERTIFICATE-----.*-----END CERTIFICATE-----");
+    public static List<X509Certificate> publicKeyFromPem(byte[] pem) {
+        Pattern pattern = Pattern.compile("(?s)-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----");
```

Comment:
Pattern compilation is expensive. So, maybe it's good to just make this pattern into a static final constant.

Reply:
In this case, it is only called once during client construction and not in any API call.
So, having it localized will help to get it garbage collected, after use.